easybuild/easyblocks/l/llvm.py

```diff
@@ -0,0 +1,80 @@
+##
+# Copyright 2020 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing LLVM, implemented as an easyblock
+
+@author: Simon Branford (University of Birmingham)
+"""
+from easybuild.easyblocks.clang import CLANG_TARGETS, DEFAULT_TARGETS_MAP
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_cpu_architecture
+
+
+class EB_LLVM(CMakeMake):
+    """
+    Support for building and installing LLVM
+    """
+
+    @staticmethod
+    def extra_options():
+        extra_vars = CMakeMake.extra_options()
+        extra_vars.update({
+            'build_targets': [None, "Build targets for LLVM (host architecture if None). Possible values: " +
+                                    ', '.join(CLANG_TARGETS), CUSTOM],
+            'enable_rtti': [True, "Enable RTTI", CUSTOM],
+        })
+        return extra_vars
+
+    def configure_step(self):
+        """
+        Install extra tools in bin/ and enable zlib; optionally enable rtti; and set the build target
+        """
+        self.cfg.update('configopts', '-DLLVM_INSTALL_UTILS=ON')
+        self.cfg.update('configopts', '-DLLVM_ENABLE_ZLIB=ON')
```

Comment:
@branfosj Should this only be done if `zlib` is included as dependency?

Reply:
Yes, that makes sense to me. Change made.